rfcs/20240311-gather-scatter-batching-dims.md

```diff
@@ -0,0 +1,456 @@
+`result_index` indices being scattered to are unique. If `unique_indices` is
+`true` but the indices being scattered to are not unique then the behavior is
+undefined.
+
+#### Inputs
+
+| Label | Name                                  | Type                                                       | Constraints                                         |
+|-------|---------------------------------------|------------------------------------------------------------|-----------------------------------------------------|
+| (I1)  | `inputs`                              | variadic number of tensors or per-tensor quantized tensors | (C1), (C2), (C4-C6), (C11), (C13), (C21), (C23-C24) |
+| (I2)  | `scatter_indices`                     | tensor of integer type                                     | (C4), (C15), (C19), (C22)                           |
+| (I3)  | `updates`                             | variadic number of tensors or per-tensor quantized tensors | (C3-C6), (C8)                                       |
+| (I4)  | `update_window_dims`                  | 1-dimensional tensor constant of type `si64`               | (C2), (C4), (C7-C8)                                 |
+| (I5)  | `inserted_window_dims`                | 1-dimensional tensor constant of type `si64`               | (C2), (C4), (C9-C11)                                |
+| (I6)  | `input_batching_dimensions`           | 1-dimensional tensor constant of type `si64`               | (C2), (C4), (C9), (C12-13), (C17-18), (C20)         |
+| (I7)  | `scatter_indices_batching_dimensions` | 1-dimensional tensor constant of type `si64`               | (C14-C18)                                           |
+| (I8)  | `scatter_dims_to_operand_dims`        | 1-dimensional tensor constant of type `si64`               | (C19-C21)                                           |
+| (I9)  | `index_vector_dim`                    | constant of type `si64`                                    | (C4), (C16), (C19), (C22)                           |
+| (I10) | `indices_are_sorted`                  | constant of type `i1`                                      |                                                     |
+| (I11) | `unique_indices`                      | constant of type `i1`                                      |                                                     |
+| (I12) | `update_computation`                  | function                                                   | (C23)                                               |
+
+#### Outputs
+
+| Name      | Type                                                       | Constraints |
+|-----------|------------------------------------------------------------|-------------|
+| `results` | variadic number of tensors or per-tensor quantized tensors | (C24-C25)   |
+
+#### Constraints
+
+* (C1) `same(shape(inputs...))`.
+* (C2) `rank(inputs[0]) = size(update_window_dims) +
+       size(inserted_window_dims) + size(input_batching_dimensions)`.
```

Comment:
`input_batching_dims`? Please check for other places as well.

Reply:
Done.